front/lib/api/assistant/agent_usage.ts

```diff
@@ -174,20 +179,46 @@ async function populateUsageIfNeeded({
   }
 }
 
-// Consider moving this to Redis if it's really slow.
+// We don't need real-time accuracy on the number of members
+// that added the agent in their list.
+// Best-effort, we cache the result for a day.
 async function getAgentInListCount(
   auth: Authenticator,
+  redis: Awaited<ReturnType<typeof redisClient>>,
+  workspaceId: string,
   agentConfiguration: LightAgentConfigurationType
 ) {
-  if (agentConfiguration.scope === "published") {
-    return getUsersWithAgentInListCount(auth, agentConfiguration.sId);
-  }
+  const { agentInListCountKey } = _getKeys({
+    agentConfigurationId: agentConfiguration.sId,
+    workspaceId,
+  });
 
-  if (agentConfiguration.scope === "workspace") {
-    return getMembersCount(auth, { activeOnly: true });
+  const cachedCount = await redis.get(agentInListCountKey);
+  if (cachedCount !== null) {
+    return parseInt(cachedCount, 10);
   }
 
-  return 0;
+  const calculateAndCacheCount = async (countPromise: Promise<number>) => {
+    const count = await countPromise;
+    await redis.set(agentInListCountKey, count.toString(), {
+      EX: 86400, // Cache for 1 day.
+    });
+    return count;
+  };
+
+  // Determine the count based on the scope and cache the result.
+  switch (agentConfiguration.scope) {
+    case "published":
+      return calculateAndCacheCount(
+        getUsersWithAgentInListCount(auth, agentConfiguration.sId)
+      );
+    case "workspace":
+      return calculateAndCacheCount(
+        getMembersCount(auth, { activeOnly: true })
+      );
+    default:
```

Comment:
here I would explicit all cases: case "global" same as workspace, case "private" is 1 (not 0, my bad :sweat_smile: ) and default assertNever

Reply:
Fixed.